### PR TITLE
Fuzz testing of execution context

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alanjo@microsoft.com>

## Description

Add a test pass that fuzzes the EBPF execution context.

## Testing

CI/CD

## Documentation

N/A

Address part of #620

### CODE REVIEW

**Review by dthaler — COMMENTED** [review]
Please update PR description to reference issue #620

File: tests/fuzz/execution_context.cpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+#include "catch_wrapper.hpp"
+#include "mock.h"
+#include "platform.h"
+#include "test_helper.hpp"
+#include "device_helper.hpp"
+#include "ebpf_core.h"
+#include <api_internal.h>
+#include <helpers.h>
+#include <bpf/libbpf.h>
```
Comment (dthaler) [inline tests/fuzz/execution_context.cpp:25]: These aren't system headers so should use "" instead of <>

File: tests/fuzz/execution_context.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+#include <filesystem>
+#include <ranges>
+#include <thread>
+#include <vector>
+
+#include "catch_wrapper.hpp"
+#include "mock.h"
+#include "platform.h"
+#include "test_helper.hpp"
+#include "device_helper.hpp"
```
Comment (dthaler) [inline tests/fuzz/execution_context.cpp:21]: sort "" headers in alphabetical order

File: tests/fuzz/execution_context.cpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        size_t minimum_request_size;
+        size_t minimum_reply_size;
+        bool async;
+        if (i % (iterations / 100) == 0) {
+            std::cout << std::dec << (i * 100) / iterations << "% completed" << std::endl;
+        }
+
+        ebpf_core_get_protocol_handler_properties(operation_id, &minimum_request_size, &minimum_reply_size, &async);
+
+        // Future - Add support for fuzzing async requests.
```
Comment (dthaler) [inline tests/fuzz/execution_context.cpp:138]: Should this be a TODO?

File: tests/fuzz/execution_context.cpp [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        if (async) {
+            continue;
+        }
+
+        request.resize(minimum_request_size + mt() % 1024);
```
Comment (dthaler) [inline tests/fuzz/execution_context.cpp:143]: Please add comments above this block explaining what it does.
I think it sets the request buffer to a random size between the minimum and 1K over the minimum, 
and then sets each byte in the request to a random value.

**Review by dthaler — DISMISSED** [review]
Looks ok, though I updated the PR description to say it only addresses part, not all, of 620, and updated 620 with a bulleted list of which this PR addresses the first one.

File: libs/platform/kernel/ebpf_platform_kernel.c [hunk 5]
```diff
@@ -9,6 +9,8 @@
 
 #include <ntstrsafe.h>
 
+bool ebpf_fuzzing_enabled = false;
```
Comment (shankarseal) [inline libs/platform/kernel/ebpf_platform_kernel.c:12]: can this be in #if CHK such that this cannot be used in retail bits by mistake?

Comment (Alan-Jowett) [reply libs/platform/kernel/ebpf_platform_kernel.c:12]: This only exists to suppress asserts etc that fire when fuzzing.

**Review by shankarseal — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Review by shankarseal — APPROVED** [review]

**Pull request merged**